src/linux/module_reader.rs

```diff
@@ -49,28 +49,48 @@ impl<'buf> ProcessMemory<'buf> {
             Self::Process(pr) => {
                 let len = std::num::NonZero::new(length as usize).ok_or_else(|| {
                     Error::ReadModuleMemory {
+                        start_address: Some(pr.start_address),
                         offset,
                         length,
                         error: nix::errno::Errno::EINVAL,
                     }
```

Comment:
Given the repetition of this rather large block it might be worth replacing it through the function with a closure, like:
```rust
let error = |start_address, error| {
    Error::ReadModuleMemory {
        start_address,
        offset,
        length,
        errror,
    }
}
```

Reply:
That was bugging me too. Will do!